service-model-generator/modules/service-model-generator-ls-extension/src/main/java/io/ballerina/servicemodelgenerator/extension/OpenApiServiceGenerator.java

```diff
@@ -132,16 +134,29 @@ public Map<String, List<TextEdit>> generateService(String typeName, String liste
         }
 
         Path mainFile = projectPath.resolve(MAIN_BAL);
-        Map<String, List<TextEdit>> textEditsMap = new HashMap<>();
+        Map<String, List<TextEdit>> textEditsMap = new LinkedHashMap<>();
         Project project = this.workspaceManager.loadProject(mainFile);
         Optional<Document> document = this.workspaceManager.document(mainFile);
         if (document.isPresent()) {
+            List<TextEdit> textEdits = new ArrayList<>();
+            ModulePartNode modulePartNode = document.get().syntaxTree().rootNode();
+
+            if (!importExists(modulePartNode, "ballerina", "http")) {
+                String importText = Utils.getImportStmt("ballerina", "http");
+                textEdits.add(new TextEdit(Utils.toRange(modulePartNode.lineRange().startLine()), importText));
+            }
+
+            if (isDefaultListenerCreationRequired) {
+                textEdits.add(new TextEdit(Utils.toRange(modulePartNode.lineRange().endLine()),
+                        ServiceModelGeneratorConstants.LINE_SEPARATOR +
+                                ServiceModelGeneratorConstants.HTTP_DEFAULT_LISTENER_STMT));
```

Comment:
If the user already has a listener with this name, wouldn't it trigger an unnecessary diagnostic? Instead, for the first generation, can we use the name generation utilities to create a unique variable name with the provided prefix (e.g., httpDefaultListener)?